docs/superpowers/specs/2026-07-01-typed-generated-column-expressions-design.md

```
@@ -0,0 +1,343 @@
+
+Generated column order should be:
+
+```text
+<name> <type> GENERATED ALWAYS AS (<expr>) <storage> [NOT NULL] [UNIQUE] ...
+```
+
+Backend-specific rendering rules:
+
+| Backend | Stored | Virtual | `SchemaExpr::concat` |
+|---|---|---|---|
+| PostgreSQL | `STORED` | Validation error initially | `a || b || c` |
```

markdownlint-cli2 (0.22.1)

[warning] Table column count
Expected: 4; Actual: 8; Too many cells, extra data will be missing

(MD056, table-column-count)